removefiles/activity.go

```diff
@@ -84,15 +85,32 @@ func removeByNames(dir, names string) (int, error) {
 		return 0, fmt.Errorf("remove: path: %q: not a directory", dir)
 	}
 
-	files := parseNames(names)
+	var res []*regexp.Regexp
+	for _, p := range split(patterns) {
+		re, err := regexp.Compile(p)
+		if err != nil {
+			return 0, fmt.Errorf("remove: pattern: %q: not a valid regular expression", p)
+		}
+		res = append(res, re)
+	}
+
+	files := split(names)
 	err = filepath.WalkDir(
 		dir,
 		func(path string, d fs.DirEntry, err error) error {
 			if err != nil {
 				return err
 			}
 
-			if slices.Contains(files, d.Name()) {
+			matches := false
+			for _, re := range res {
```

Comment:
I just noticed this on re-reading the code: it's less expensive to check if `slices.Contains(files, d.Name())` matches than doing the regex matches, so I think it would do the simple name matching first.

Reply:
Done in latest commit.